dotnet/src/Connectors/Connectors.OpenAI/Core/OpenAIFunction.cs

```diff
@@ -127,52 +139,153 @@ internal OpenAIFunction(
     /// <see cref="ChatTool"/> representation.
     /// </summary>
     /// <returns>A <see cref="ChatTool"/> containing all the function information.</returns>
-    public ChatTool ToFunctionDefinition()
+    [Obsolete("Use the overload that takes a boolean parameter instead.")]
+    public ChatTool ToFunctionDefinition() => this.ToFunctionDefinition(false);
+
+    /// <summary>
+    /// Converts the <see cref="OpenAIFunction"/> representation to the OpenAI SDK's
+    /// <see cref="ChatTool"/> representation.
+    /// </summary>
+    /// <returns>A <see cref="ChatTool"/> containing all the function information.</returns>
+    public ChatTool ToFunctionDefinition(bool allowStrictSchemaAdherence)
```

Comment:
Optional and default to false?

Reply:
this is what I originally had before the public API surface compatibility called me out. Adding an optional parameter is technically a binary breaking change. Hence the obsolete approach.